core/src/Kokkos_View.hpp

```diff
@@ -1402,22 +1402,13 @@ class View : public ViewTraits<DataType, Properties...> {
                        typename traits::array_layout> const& arg_layout,
       check_input_args check_args = check_input_args::no)
       : m_track(), m_map() {
-    // Append layout and spaces if not input
-    using alloc_prop_input = Impl::ViewCtorProp<P...>;
-
-    // use 'std::integral_constant<unsigned,I>' for non-types
-    // to avoid duplicate class error.
-    using alloc_prop = Impl::ViewCtorProp<
-        P...,
-        std::conditional_t<alloc_prop_input::has_label,
-                           std::integral_constant<unsigned int, 0>,
-                           std::string>,
-        std::conditional_t<alloc_prop_input::has_memory_space,
-                           std::integral_constant<unsigned int, 1>,
-                           typename traits::device_type::memory_space>,
-        std::conditional_t<alloc_prop_input::has_execution_space,
-                           std::integral_constant<unsigned int, 2>,
-                           typename traits::device_type::execution_space>>;
+    // Append layout and spaces if not input, split for MSVC
+    using alloc_prop_tmp =
+        decltype(Impl::add_properties(arg_prop, std::string{}));
+    using alloc_prop = decltype(
+        Impl::add_properties(std::declval<alloc_prop_tmp>(),
+                             typename traits::device_type::memory_space{},
+                             typename traits::device_type::execution_space{}));
```

Comment:
Why does this go through defining and extending `alloc_prop_tmp`? It looks like all of these properties can be added in one shot, like you did in `OffsetView`

Reply:
I tried to say in the comment above that MSVC was giving me errors when defining the alias in one shot.

Reply:
Could you add a comment to that effect in the code? Otherwise, people may try to refactor it and be disappointed

Reply:
Ditto the suggestion above of creating `prop_copy` and using `decltype(prop_copy)` from above - does that also avoid the MSVC errors?